app/controllers/forms_controller.rb

```diff
@@ -41,8 +41,17 @@ def calculate_early_result
     if last_tag_in_group?(:gross_income)
       cfe_result = CfeService.result(session_data, Steps::Helper.completed_steps_for(session_data, step))
       session_data["early_result"] = { "result" => cfe_result.gross_income_result,
-                                       "gross_income_excess" => cfe_result.gross_income_excess }
-      # add analytics tracking here? completed user journeys
+                                       "gross_income_excess" => cfe_result.gross_income_excess,
+                                       "type" => "gross_income" } # I think we have to set this here so we have it in the session, when calling the `JourneyLoggerService`
+      track_completed_journey_for_early_result
     end
   end
+
+  def track_completed_journey_for_early_result
```

Comment:
have you thought about/tried passing down the CFE result stored in `cfe_result` to the tracking method? That might prevent the need to make the CFE call again. Variables are a sort of storage - once they've been set, they store that information.

So you'd do something like track_early_result_completed_journey(cfe_result) and then use/manipulate cfe_result to get what you need.

Reply:
Yes, have tried passing down the CFE result stored in `cfe_result` to the tracking method. 

I think the issue is that they are two different method calls to the `CfeService`. I need the `.call` in `track_completed_journey_for_early_result` instead of `.result`